From a808e6ef4ebc96b8f66ad13bd3caab32aa05505d Mon Sep 17 00:00:00 2001
From: Jean Delvare <jdelvare@suse.de>
Date: Wed, 16 Oct 2019 15:35:42 +0200
Subject: [PATCH 15/40] Typo

---
 dmidecode.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dmidecode.c b/dmidecode.c
index 68bfa45..bdf1185 100644
--- a/dmidecode.c
+++ b/dmidecode.c
@@ -1901,7 +1901,7 @@ static const char *dmi_slot_current_usage(u8 code)
 
 static const char *dmi_slot_length(u8 code)
 {
-	/* 7.1O.4 */
+	/* 7.10.4 */
 	static const char *length[] = {
 		"Other", /* 0x01 */
 		"Unknown",
-- 
2.26.2

